components/director/pkg/webhook/types.go

```diff
@@ -174,7 +174,9 @@ func parseTemplate(tmpl *string, data interface{}, dest interface{}) error {
 	if err = t.Execute(res, data); err != nil {
 		return err
 	}
-	if err = json.Unmarshal(res.Bytes(), dest); err != nil {
+	resBytes := bytes.ReplaceAll(res.Bytes(), []byte("\"<nil>\""), []byte("\"\""))
+	resBytes = bytes.ReplaceAll(resBytes, []byte("<nil>"), []byte("null"))
```

Comment:
Could you please add two comments for both replace scenarios? E.g. the first one is when the field in the go code is a pointer(nil/empty) and we have explicit quotes in the templates resulting in the `"<nil>"` but the second case is when we don't have quotes in the template and we have only `<nil>`?